### ISSUE TITLE
UP035: No violations when importing an entire module and using deprecated/moved class/method

### ISSUE THREAD

**Issue by einarwar** [astral-sh/ruff#7526]
When explicitly importing deprecated/moved classes/methods UP035 works as expected. Example snippet that will raise a UP035: 
```python
from typing import Mapping
# Import from `collections.abc` instead: `Mapping`Ruff[UP035](https://beta.ruff.rs/docs/rules/deprecated-import)
```

However, when importing the entire module from where the deprecated/moved submodule originally was, no error is raised. Example snippet:
```python
import typing
x: typing.Mapping[str, int] = {}
# No error is raised
```

I would think the expected behaviour would be to raise an UP035 violation on the line where the deprecated/moved class is defined, in the above case line 2. 

Settings:
Ruff 0.0.290
Ruff-VSCode v2023.38.0 with `fromEnvironment` as import-strategy
Enabled UP in pyproject.toml

**Comment by jerr0328** [comment]
Seems like it's also mentioned in #3936, I was surprised that ruff didn't catch this but I guess it's a bit trickier to catch and fix.

**Comment by MichaReiser** [comment]
Flagging full qualified name usages does make sense to me. The only issue is that the rule only flags imports, not usages, which seems an issue with how the rule is designed
